### ISSUE TITLE
Enable both manual and automatic HTTPS/SSL mode

### ISSUE THREAD

**Issue by nermin99** [vercel/serve#627]
Why was support(#274) for automatic HTTPS removed? As per #520 and #511 it is understandable that users might want to use manual certs and keys, **but** there is still, no even more, use cases for automatic SSL/HTTPS. One of the biggest reasons why `serve` is so popular is because it's a one-line command to literally "serve" your application. If a user wants or for some reason needs HTTPS they don't wanna have to go through a whole process of installing another tool just to get it.

Personally I stumbled upon `serve` when building my create-react-app and wanted to "serve" it. But because I use service workers and push notifications I need to serve it over HTTPS so I can test on other devices on the network. Sure I can get HTTPS for my self as per #520, but my co-workers can't, since they might not have their own manual certs and keys... you see the problem?

**There is no reason why both manual and automatic HTTPS can't be supported.**

**Comment by TimDaub** [comment]
> For now I'm running `serve` at version 6 instead (`npx serve@6 --ssl ...`) and it's working well for my particular use case

lovely :) Thanks for pointing out that version.

**Comment by TimDaub** [comment]
I fix and published vercel/serve@6.5.8 as `npm i ssl-serve`.
Repo: https://github.com/TimDaub/ssl-serve

**Comment by warren-bank** [comment]
_shameless self-promotion alert_:<br>this feature is available in my [`@warren-bank/serve`](https://github.com/warren-bank/node-serve) fork of `serve`

**Comment by AndyBitz** [comment]
@gavinr-maps Yes! That sounds like a good idea. Functionality wise I can imagine something like we have for Next.js `next dev --experimental-https`.

**Comment by rlidwka** [comment]
Is there any option in HTTP 2 or HTTP 3 for opportunistic encryption?

I want `serve` to be encrypted by default (e.g. if I need to use it to share files to my phone over wifi). But I don't want the browser to scream at me for doing so.
